File: test/testspringboot/src/main/java/pers/acp/test/application/base/BaseRepository.java
package pers.acp.test.application.base;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;

import java.io.Serializable;

/**
 * @author zhangbin by 2018-1-15 16:59
 * @since JDK1.8
 */
public interface BaseRepository<T, ID extends Serializable> extends JpaSpecificationExecutor<T>, JpaRepository<T, ID> {
}

File: test/testscala/src/main/scala/com/test/scala/base/BaseRepository.scala
package com.test.scala.base

import org.springframework.data.jpa.repository.{JpaRepository, JpaSpecificationExecutor}
import org.springframework.data.repository.NoRepositoryBean

/**
  * @author zhangbin by 02/05/2018 17:36
  * @since JDK 11
  */
@NoRepositoryBean
trait BaseRepository[T, ID] extends JpaSpecificationExecutor[T] with JpaRepository[T, ID]